Core/GDCore/IDE/Project/ArbitraryObjectsWorker.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#ifndef ARBITRARYOBJECTSWORKER_H
+#define ARBITRARYOBJECTSWORKER_H
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "GDCore/String.h"
+namespace gd {
+class Object;
+class ObjectsContainer;
+class BehaviorContent;
+}  // namespace gd
+
+namespace gd {
+
+/**
+ * \brief ArbitraryObjectsWorker is an abstract class used to browse objects
+ * (and behaviors) and do some work on them. Can be used to implement
+ * refactoring for example.
+ *
+ * \ingroup IDE
+ */
+class GD_CORE_API ArbitraryObjectsWorker {
+ public:
+  ArbitraryObjectsWorker(){};
+  virtual ~ArbitraryObjectsWorker();
+
+  /**
+   * \brief Launch the worker on the specified events list.
+   */
+  void Launch(gd::ObjectsContainer& objects) { VisitObjectContainer(objects); };
+
+ private:
+  void VisitObjectContainer(gd::ObjectsContainer& objects);
+  void VisitObject(gd::Object& object);
+  void VisitBehavior(gd::BehaviorContent& instruction);
+
+  /**
+   * Called to do some work on an object container.
+   */
+  virtual void DoVisitObjectContainer(gd::ObjectsContainer& objects){};
+
+  /**
+   * Called to do some work on an object.
+   */
+  virtual void DoVisitObject(gd::Object& object){};
+
+  /**
+   * Called to do some work on a behavior.
+   */
+  virtual void DoVisitBehavior(gd::BehaviorContent& instruction);
```

Comment:
nit: behavior

Reply:
```suggestion
  virtual void DoVisitBehavior(gd::BehaviorContent& instruction) {};
```

otherwise you'll get the  undefined reference.
Or you choose not to provide a default implementation and use `= 0` to have a pure virtual function (i.e: your class is now an abstract class). Might actually be better to avoid subclasses to avoid overriding something?

Reply:
Well all workers will not always want to visit behaviors or ObjectContainers, maybe it's only interested in objects, it would be pretty verbose and bloating to have to write all the methods out in sub-classes.

Reply:
Fair enough to keep an empty implementation then!